packages/backend/src/util/request.ts

```diff
@@ -57,3 +62,11 @@ function getPort(url: URL) {
 function getPortFromProtocol(protocol: string) {
   return PROTOCOL_TO_PORT_MAPPING[protocol];
 }
+
+function getFirstValueFromHeaderValue(value?: string | null) {
+  return value?.split(',')[0]?.trim() || '';
+}
+
+function getProtocolVerb(protocol: string) {
+  return protocol?.replace(/:$/, '') || '';
```

Comment:
Just curious, is there any protocol with more than 1 `:`?

Reply:
I believe not based on this RFC: https://www.rfc-editor.org/rfc/rfc1738#section-5

> genericurl     = scheme ":" schemepart